mixnet/protocol/connection.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import abc
+
+from framework import Framework, Queue
+
+
+class SimplexConnection(abc.ABC):
```

Comment:
The `SimplexConnection` is the minimum unit, and a single queue is going to be used to implement it actually. And, we have the `DuplexConnection` class, which has two `SimplexConnection`s. I think it's good to have the `SimplexConnection` as the minimum unit, so that we can extend it for special features such as GTR and noise (as I did in the `MixSimplexConneciton` class).

The `BroadcastConnection` that you suggested is probably the same as `DuplexConnection`? or are you suggesting something else (maybe merging multiple `SimplexConnections` into a single `BroadcastConnection`)?